cli/cmdlineparser.cpp

```diff
@@ -59,6 +59,10 @@
 #ifdef HAVE_RULES
 // xml is used for rules
 #include "xml.h"
+
+#ifdef _MSC_VER
+#pragma message("The usage of rules is deprecated and will be removed in a future Cppcheck version. Please use addons instead.")
+#endif
```

Comment:
This is for the case when you compile it through the provided Visual Studio project which utilized neither CMake nor make.